From 374d9df9307d6cb1572725982307f1e6e57ea683 Mon Sep 17 00:00:00 2001
From: Minghuan Lian <Minghuan.Lian@freescale.com>
Date: Fri, 3 Jul 2015 14:57:08 +0800
Subject: [PATCH 056/124] pci/msi: fix checking msi list

Function list_empty should be used to check a list other
than list ponter.

Signed-off-by: Minghuan Lian <Minghuan.Lian@freescale.com>
---
 drivers/pci/msi.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/pci/msi.c b/drivers/pci/msi.c
index f535fee..352e828 100644
--- a/drivers/pci/msi.c
+++ b/drivers/pci/msi.c
@@ -908,7 +908,7 @@ void pci_msi_shutdown(struct pci_dev *dev)
 	if (!pci_msi_enable || !dev || !dev->msi_enabled)
 		return;
 
-	BUG_ON(dev_to_msi_list(&dev->dev));
+	BUG_ON(list_empty(dev_to_msi_list(&dev->dev)));
 	desc = first_msi_entry(&dev->dev);
 
 	msi_set_enable(dev, 0);
-- 
2.1.0.27.g96db324

